shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -583,119 +521,114 @@ - (void)startProfiler {
+                                           codec:[FlutterStandardMethodCodec sharedInstance]];
+
+  self.lifecycleChannel =
+      [[FlutterBasicMessageChannel alloc] initWithName:@"flutter/lifecycle"
+                                       binaryMessenger:self.binaryMessenger
+                                                 codec:[FlutterStringCodec sharedInstance]];
+
+  self.systemChannel =
+      [[FlutterBasicMessageChannel alloc] initWithName:@"flutter/system"
+                                       binaryMessenger:self.binaryMessenger
+                                                 codec:[FlutterJSONMessageCodec sharedInstance]];
+
+  self.settingsChannel =
+      [[FlutterBasicMessageChannel alloc] initWithName:@"flutter/settings"
+                                       binaryMessenger:self.binaryMessenger
+                                                 codec:[FlutterJSONMessageCodec sharedInstance]];
+
+  self.keyEventChannel =
+      [[FlutterBasicMessageChannel alloc] initWithName:@"flutter/keyevent"
+                                       binaryMessenger:self.binaryMessenger
+                                                 codec:[FlutterJSONMessageCodec sharedInstance]];
+
+  self.textInputPlugin = [[FlutterTextInputPlugin alloc] initWithDelegate:self];
+  self.textInputPlugin.indirectScribbleDelegate = self;
+  [self.textInputPlugin setUpIndirectScribbleInteraction:self.viewController];
+
+  self.undoManagerPlugin = [[FlutterUndoManagerPlugin alloc] initWithDelegate:self];
+  self.platformPlugin = [[FlutterPlatformPlugin alloc] initWithEngine:self];
+
+  self.restorationPlugin =
+      [[FlutterRestorationPlugin alloc] initWithChannel:self.restorationChannel
+                                     restorationEnabled:self.restorationEnabled];
+  self.spellCheckPlugin = [[FlutterSpellCheckPlugin alloc] init];
+
+  self.screenshotChannel =
+      [[FlutterMethodChannel alloc] initWithName:@"flutter/screenshot"
+                                 binaryMessenger:self.binaryMessenger
+                                           codec:[FlutterStandardMethodCodec sharedInstance]];
+
+  [self.screenshotChannel
       setMethodCallHandler:^(FlutterMethodCall* _Nonnull call, FlutterResult _Nonnull result) {
-        if (!(weakSelf.get() && weakSelf.get()->_shell && weakSelf.get()->_shell->IsSetup())) {
+        FlutterEngine* strongSelf = weakSelf;
+        if (!(strongSelf && strongSelf->_shell && strongSelf->_shell->IsSetup())) {
           return result([FlutterError
               errorWithCode:@"invalid_state"
                     message:@"Requesting screenshot while engine is not running."
                     details:nil]);
         }
         flutter::Rasterizer::Screenshot screenshot =
-            [weakSelf.get() screenshot:flutter::Rasterizer::ScreenshotType::SurfaceData
-                          base64Encode:NO];
+            [weakSelf screenshot:flutter::Rasterizer::ScreenshotType::SurfaceData base64Encode:NO];
```

Comment:
Nit: technically it doesn't matter, but generally with the weakSelf/strongSelf pattern you only use `strongSelf` once it's created.

Reply:
Oh good catch -- I wonder if I did the two blocks in different passes, but totally miss ed spotting this one. Done. Thanks!